File: 0xxx/04xx/040x/0409.ts
function longestPalindrome(s: string): number {
  const length = s.length
  const hash: { [key: string]: number } = {}
  let result: number = 0

  for (let i = 0; i < length; i++) {
    if (hash[s[i]]) {
      hash[s[i]]--
      result += 2
    } else {
      hash[s[i]] = 1
    }
  }

  return result === length ? result : result + 1
}

File: 0xxx/04xx/040x/0409.py
class Solution:
  def longestPalindrome(self, s: str) -> int:
    result = 0
    recorder = {}

    for letter in s:
      if recorder.get(letter, 0) == 0:
        recorder[letter] = 1
      else:
        recorder[letter] -= 1
        result += 2

    if len(s) > result:
      result += 1
      
    return result